### PR TITLE
Fix 'create_result_query' ordering

### PR DESCRIPTION
**Pull request by dreadatour**
Tiny fix for unused code found while working on another PR.

### CODE REVIEW

File: src/datachain/query/dataset.py [hunk 1]
```diff
@@ -666,13 +666,6 @@ def create_udf_table(self, query: Select) -> "Table":
     def create_result_query(
         self, udf_table, query: Select
     ) -> tuple[QueryGeneratorFunc, list["sqlalchemy.Column"]]:
-        if not query._order_by_clauses:
-            # if we are not selecting all rows in UDF, we need to ensure that
-            # we get the same rows as we got as inputs of UDF since selecting
-            # without ordering can be non deterministic in some databases
-            c = query.selected_columns
-            query = query.order_by(c.sys__id)
```
Comment (dreadatour) [inline src/datachain/query/dataset.py:674]: This `query` variable is not used after reassigning.

File: src/datachain/query/dataset.py [hunk 2]
```diff
@@ -682,7 +675,15 @@ def q(*columns):
...
             # Columns for the generated table.
             cols = [c for c in udf_table_cols if c.name in names]
-            return sqlalchemy.select(*cols).select_from(udf_table_query)
+            res = sqlalchemy.select(*cols).select_from(udf_table_query)
+
+            if query._order_by_clauses:
+                # if we are not selecting all rows in UDF, we need to ensure that
+                # we get the same rows as we got as inputs of UDF since selecting
+                # without ordering can be non deterministic in some databases
+                res = res.order_by(udf_table_query.c.sys__id)
```
Comment (dreadatour) [inline src/datachain/query/dataset.py:684]: Move `order_by` here, [same](https://github.com/iterative/datachain/blob/main/src/datachain/query/dataset.py#L633-L636) as in `UDFSignal.create_result_query` method.

Comment (mattseddon) [reply src/datachain/query/dataset.py:684]: I am not against this change but PTAL at this: https://github.com/iterative/datachain/issues/477#issuecomment-2387274525. It looks unlikely we'll be able to support this behaviour in BQ.

**Review by ilongin — APPROVED** [review]

File: src/datachain/query/dataset.py [hunk 3]
```diff
@@ -682,7 +675,15 @@ def q(*columns):
             names = {c.name for c in columns}
             # Columns for the generated table.
             cols = [c for c in udf_table_cols if c.name in names]
-            return sqlalchemy.select(*cols).select_from(udf_table_query)
+            res = sqlalchemy.select(*cols).select_from(udf_table_query)
+
+            if query._order_by_clauses:
```
Comment (mattseddon) [inline src/datachain/query/dataset.py:680]: qq - how come you are switching from `if not query._order_by_clauses:` to `if query._order_by_clauses:`?

**Comment by dreadatour** [comment]
Closing this issue because we decided to do not preserve ordering within dataset.

File: src/datachain/query/dataset.py (hunk 3 above)
Comment (dreadatour) [reply src/datachain/query/dataset.py:680]: To be honest I am not sure why it was `not` there, may be it is because of previous refactoring attempts.

**Pull request closed without merging**